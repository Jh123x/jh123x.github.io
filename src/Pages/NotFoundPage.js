import styled from "styled-components";

const timeout = 3000;

/**
 * NotFoundPage is displayed when the path is not found.
 * @returns {React.JSX.Element}
 */
const NotFoundPage = () => {
  setTimeout(() => {
    window.location.replace("/");
  }, timeout);
  return (
    <HomePageStyled>
      <div className="typography">
        <h1>404 Page not found</h1>
        <p>
          This page does not / no longer exists.
          <br />
          You will be redirected in {timeout / 1000} seconds
        </p>
      </div>
    </HomePageStyled>
  );
};

const HomePageStyled = styled.header`
  width: 100%;
  height: 100vh;
  position: relative;

  .typography {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    width: 80%;

    .icons {
      display: flex;
      justify-content: center;
      margin-top: 1rem;
      .icon {
        border: 2px solid var(--border-color);
        display: flex;
        align-items: center;
        justify-content: center;
        border-radius: 50%;
        transition: all 0.4s ease-in-out;
        cursor: pointer;
        &:hover {
          border: 2px solid var(--primary-color);
          color: var(--primary-color);
        }
        &:not(:last-child) {
          margin-right: 1rem;
        }
        svg {
          margin: 0.5rem;
        }
      }

      .i-youtube {
        &:hover {
          border: 2px solid red;
          color: red;
        }
      }
      .i-github {
        &:hover {
          border: 2px solid #5f4687;
          color: #5f4687;
        }
      }
    }
  }
`;

export default NotFoundPage;
