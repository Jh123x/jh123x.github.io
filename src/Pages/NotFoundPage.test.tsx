import * as React from "react";
import NotFoundPage from "./NotFoundPage";
import { describe, expect, test } from "@jest/globals";

describe("Render NotFoundPage", () => {
  test("renders NotFoundPage component", () => {
    const notFound = <NotFoundPage />;
    expect(notFound).toMatchSnapshot();
  });
});
