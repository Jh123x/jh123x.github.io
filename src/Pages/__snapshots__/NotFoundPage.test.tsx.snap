// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotFoundPage renders NotFoundPage component 1`] = `<NotFoundPage />`;
